OTAnalytics/plugin_parser/otvision_parser.py

```diff
@@ -266,13 +304,25 @@ def __parse_otdet_version(self, content: dict) -> Version:
         version = content[ottrk_format.METADATA][ottrk_format.OTDET_VERSION]
         return Version.from_str(version)
 
+    # TODO find better name instead of name duplication?
     def __fix_metadata(self, content: dict) -> dict:
-        metadata = content[ottrk_format.METADATA]
+        content[ottrk_format.METADATA] = self.fix_metadata(
+            metadata=content[ottrk_format.METADATA]
+        )
+        return content
+
+    # TODO Review: fixing metadata alone can be reused in streaming_parser,
+    # TODO hence I made it public. This might also apply to fixing
+    # TODO single detection data, when they are parsed lazy (but currently not used).
+    def fix_metadata(self, metadata: dict) -> dict:
+        """
+        Fix formate changes from older ottrk metadata
+        format versions to the current version.
+        """
         current_version = Version.from_str(metadata[ottrk_format.OTTRK_VERSION])
```

Comment:
@briemla Review: fixing metadata alone can be reused in streaming_parser, hence I made it public. 
This might also apply to fixing single detection data, when they are parsed lazy (but currently not used).

Reply:
Fixed in commit 4aaf9a5, metadatafixer can now define whether they are based on otdet or ottrk format version